### PR TITLE
perlPackages.Apprainbarf: init at 1.4

### PR DESCRIPTION
**Pull request by saschagrunert**
###### Motivation for this change
A perl package I love, please let this get into nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -16716,6 +16716,21 @@ let
...
   };
 
+  Rainbarf = buildPerlModule {
+    pname = "Rainbarf";
+    version = "1.4";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/S/SY/SYP/App-rainbarf-1.4.tar.gz";
+      sha256 = "4f139ad35faaf2de0623dc0bb1dd89fa5a431e548bfec87dee194cf0e25cc97d";
+    };
+    buildInputs = [ ModuleBuild ];
```
Comment (conferno) [inline pkgs/top-level/perl-packages.nix:16726]:
```suggestion

```
`buildPerlModule` implicitly includes `ModuleBuild`

**Review by conferno — CHANGES_REQUESTED** [review]

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -16716,6 +16716,21 @@ let
     };
   };
 
+  Rainbarf = buildPerlModule {
+    pname = "Rainbarf";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:16720]:
```suggestion
  Apprainbarf = buildPerlModule {
    pname = "Apprainbarf";
```
The attribute name and pname should follow the conventions. Also use `perlPackages.Apprainbarf` in the commit message and PR title.

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -16716,6 +16716,21 @@ let
...
+    version = "1.4";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/S/SY/SYP/App-rainbarf-1.4.tar.gz";
+      sha256 = "4f139ad35faaf2de0623dc0bb1dd89fa5a431e548bfec87dee194cf0e25cc97d";
+    };
+    buildInputs = [ ModuleBuild ];
+    meta = {
+      homepage = https://github.com/creaktive/rainbarf;
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:16728]:
```suggestion
      homepage = "https://github.com/creaktive/rainbarf";
```

Pls quote urls

**Review by stigtsp — CHANGES_REQUESTED** [review]
To make `rainbarf` run on darwin:
```nix
    nativeBuildInputs = stdenv.lib.optional stdenv.isDarwin shortenPerlShebang;
    postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
      shortenPerlShebang $out/bin/rainbarf
    '';
```

**Comment by saschagrunert** [comment]
Thank you for the kind review! :pray: I added the suggested changes, please take a look again. :)

**Review by stigtsp — CHANGES_REQUESTED** [review]
Pls change commit message to `perlPackages.Apprainbarf: init at 1.4`

**Review by stigtsp — APPROVED** [review]
LGTM :) Thx for the PR

(To be pedantic: maybe update the PR name as well)

**Pull request merged**